core/src/main/java/org/kiwix/kiwixmobile/core/main/CompatFindActionModeCallback.kt

```diff
@@ -0,0 +1,196 @@
+  fun setWebView(webView: WebView?) {
+    this.webView = requireNotNull(webView) {
+      "WebView supplied to CompatFindActionModeCallback cannot be null"
+    }
+    findResultsTextView.visibility = View.VISIBLE
+    this.webView?.setFindListener { activeMatchOrdinal: Int, numberOfMatches: Int, _: Boolean ->
+      findResultsTextView.text = when {
+        editText.text.toString().isEmpty() -> ""
+        numberOfMatches == 0 -> "0/0"
+        else -> "${activeMatchOrdinal + 1}/$numberOfMatches"
+      }
+    }
+  }
+
+  // Move the highlight to the next match.
+  // If true, find the next match further down in the document.
+  // If false, find the previous match, up in the document.
+  private fun findNext(next: Boolean) {
+    requireNotNull(webView) {
+      "No WebView for CompatFindActionModeCallback::findNext"
+    }.findNext(next)
+  }
+
+  // Highlight all the instances of the string from mEditText in mWebView.
+  fun findAll() {
+    requireNotNull(webView) {
+      "No WebView for CompatFindActionModeCallback::findAll"
+    }
+    val textToFind: CharSequence? = editText.text
+    if (textToFind?.isNotEmpty() == true) {
+      webView?.findAllAsync("$textToFind")
+
+      // Enable word highlighting with reflection
+      try {
+        WebView::class.java.declaredMethods
+          .firstOrNull { it.name == "setFindIsUp" }
+          ?.let { method ->
+            method.isAccessible = true
+            method.invoke(webView, true)
+          }
+      } catch (exception: Exception) {
+        Log.e(tag, "Exception in findAll", exception)
+      }
+    } else {
+      webView?.clearMatches()
+      webView?.findAllAsync("")
+    }
+  }
+
+  // Show on screen keyboard
+  fun showSoftInput() {
+    // wait for any hidden show/hide processes to finish
+    editText.postDelayed({
+      editText.requestFocus()
+      // show the keyboard
+      input.showSoftInput(editText, 0)
+    }, 100)
+  }
+
+  override fun onClick(v: View) {
+    findNext(true)
+  }
+
+  override fun onCreateActionMode(mode: ActionMode, menu: Menu): Boolean {
+    mode.customView = customView
+    mode.menuInflater.inflate(R.menu.menu_webview, menu)
+    actionMode = mode
+    val edit = editText.text
+    Selection.setSelection(edit, edit.length)
+    editText.requestFocus()
+    return true
+  }
+
+  override fun onDestroyActionMode(mode: ActionMode) {
+    actionMode = null
+    isActive = false
+    webView?.clearMatches()
+    input.hideSoftInputFromWindow(webView?.windowToken, 0)
+  }
+
+  override fun onPrepareActionMode(mode: ActionMode, menu: Menu): Boolean = false
+
+  override fun onActionItemClicked(mode: ActionMode, item: MenuItem): Boolean {
+    requireNotNull(webView) {
+      "No WebView for CompatFindActionModeCallback::onActionItemClicked"
+    }
+    input.hideSoftInputFromWindow(webView?.windowToken, 0)
+    when (item.itemId) {
+      R.id.find_prev -> {
```

Comment:
also these brackets can go, sorry for the scattered review, should have been more focused

Reply:
No issues f5a734f
I should have handled it myself.